src/metpy/plots/declarative.py

```diff
@@ -699,6 +732,60 @@ def _set_need_redraw(self, _):
         """Watch traits and set the need redraw flag as necessary."""
         self._need_redraw = True
 
+    @staticmethod
+    def _zoom_extent(extent, zoom):
+        """Calculate new bounds for zooming in or out of a given extent.
+
+        ``extent`` is given as a tuple with four numeric values, in the same format as the
+        ``area`` trait.
+
+        If ``zoom`` = 0, the extent will not be changed from what was provided to the method
+        If ``zoom`` > 0, the returned extent will be smaller (zoomed in)
+        If ``zoom`` < 0, the returned extent will be larger (zoomed out)
+
+        """
+        # Measure the current extent
+        center_lon = (extent[0] + extent[1]) / 2
+        center_lat = (extent[2] + extent[3]) / 2
+        lon_range = extent[1] - extent[0]
+        lat_range = extent[3] - extent[2]
+
+        # Transforming zoom by e^-x prevents multiplication by zero or a negative number below
+        zoom_multiplier = np.exp(-0.5 * zoom)
+
+        # Calculate "width" and "height" of new, zoomed extent
+        new_lon_range = lon_range * zoom_multiplier
+        new_lat_range = lat_range * zoom_multiplier
+
+        # Calculate bounds for new, zoomed extent with new "width" and "height"
+        new_west_lon = center_lon - 0.5 * new_lon_range
+        new_east_lon = center_lon + 0.5 * new_lon_range
```

Comment:
You can eliminate the ranges/centers and write this as:

```suggestion
        new_west_lon = 0.5 * ((1 + zoom) * extent[0] + (1 - zoom) * extent[1])
        new_east_lon = 0.5 * ((1 + zoom) * extent[1] + (1 - zoom) * extent[0])
```
(and likewise for lat). I'm guess that's simpler? Thoughts?